### DISCUSSION TITLE
Análisis de requerimientos

### DISCUSSION THREAD

**Discussion by GabrielaEsquivel** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#70] [Ideas]
¡Hola, compañeros!
¿Cómo están? Buenas. Vengo por acá a compartir solo un poquito de mi conocimiento. Estudié ing. informática y tuve una materia dedicada al diseño de sistemas y eso incluyó los requerimientos. 
La toma de requerimientos para una app/sistema puedes hacerla de distintas maneras, ya sea que tengas una conversación con tu cliente en donde él te diga algo como:
_Quiero una app móvil que tenga colores azules y les permita a las personas registrarse.. bla bla._
También puedes entrevistarlo y hacerle preguntas cómo: _¿Qué es lo que quieres que tu app haga? ¿Los usuarios se pueden registrar? ¿Qué datos tomas de cada persona? etc, etc._

Una vez que obtienes ese resumen o historia de usuario (pueden googlearlo así como "historia de usuario") lo que deberías o hacer es documentarlo. 
Existen distintos formatos para hacer esto. El que yo utilicé luce así: 
https://user-images.githubusercontent.com/99062142/155241817-79360821-9f5d-4f37-afb1-8adee0dfadfe.png 

En donde pongo el título del requerimiento (siempre es bueno enumerarlos), hago una descripción de este y le doy una importancia, esto me ayudará a saber si debería empezar a hacerlo o no, qué tanto cuidado debo de prestarle, etc, es más para un control interno, así lo veo yo. Te puedes guiar de acuerdo a lo que tu cliente te diga, si es muy importante, o porque simplemente puedas deducirlo.
Después hay un apartado que tiene "Validación", técnicamente en este rubro tú describes 1, cómo es que podrá ser comprobable que implementaste el requisito (medible), qué es lo que se verá cuando lo hayas implementado.
También pondrás cómo es que lo vas a lograr, en base a qué tú puedes saber que puedes cumplir dicho requisito (alcanzable) y por último alguna nota importante.
A esta tabla tú le puedes añadir más campos, tal como hice aquí:
https://user-images.githubusercontent.com/99062142/155242628-ac8bc3e2-a0c5-41e0-a1f0-c6673288f058.png

Puedes hacer una mezcla de los recursos que el MC proporcionó, con lo que encuentres en internet. y más.

Y si te pregunta es... ¿cómo sé cuál es un requerimiento?
Pues bueno, tomando en cuenta que el cliente te ha diicho; 
**_Quiero una app móvil que tenga colores azules y les permita a las personas registrarse.. bla bla._**
Un requisito sería: que el sistema tenga los colores azules que tenga
Otro requisito sería: que personas puedan registrarse en la plataforma
¿Cómo sabes que son requisitos? Porque son características que tu cliente te ha dicho que quiere o necesita que el producto tenga. 

Espero que lo que haya podido compartir por aquí sea de ayuda.
Cualquier cosa en la que pueda apoyar, escríbanme, responderé ASAP.

**Comment by GabrielaEsquivel** [comment]
Otro punto que me gustaría mencionar:
Si quieres aprender más de los requisitos, puedes buscar sus distintas categorías, OJO, recomiendo esto solamente si te sientes seguro de que no te vayas a confundir más 🥴  por que sí pasa que uno se confunde con tanto; por ejemplo, están los requisitos funcionales, de evolución, de soporte, de calidad, de diseño de UI, etc.
Ahora sí creo que eso es todo. Gracias por leer.

**Reply by erykzon** [reply]
Muchas gracias por compartir esta info.

**Comment by LuceroLuciano** [comment]
Gracias por compartir. 
Saludos.

**Reply by CarlaPerez22** [reply]
Muchas gracias por compartir!

Crees que sea adecuado agregar casos de uso o con las historias estaría bien?

**Reply by GabrielaEsquivel** [reply]
A mí parecer siempre es bueno tener otro apoyo para el análisis de requerimientos, depende mucho de qué se trate y que tanto detalle necesite pienso yo, si crees que para un determinado requisito vendría bien un caso de uso, adelante :D

**Comment by GloriaCoralCerecedo** [comment]
@GabrielaEsquivel 

Hola, igual yo tengo algo de conocimiento, gracias por los datos. Pero te tengo unas preguntas. ¿Tu hiciste el modelamiento de negocio y la fase de planeación?

**Reply by GabrielaEsquivel** [reply]
Hola, Gloria, el modelado de negocio sí que lo he hecho, de hecho ando documentándolo, la planeación no, PERO sí he hecho planeaciones de las cuales tengo ejemplos.

**Reply by GloriaCoralCerecedo** [reply]
Oh vale tengo entendido que es como explicar en esquemas como funciona el despacho de abogados agregando la automatización no¿? El de planeación igual creo que no sera muy necesario. O puede que si. Igual de la plantilla del MC quite algunas cosas como las firmas

**Reply by GabrielaEsquivel** [reply]
Sí, pero, bueno tenemos distintos diagramas para documentar o diseñar el comportamiento de un sistema, tales como los de actividad, los BPMN (modelado de negocio), de secuencia, de ciclos o estados, entre otros.
En el caso de los BPMN (que al igual que todos tiene una manera de hacerlo, es decir, tiene figuras, líneas, etc preestablecidos) simula la lógica de negocio. Es decir, por ejemplo, si fueras a diseñar un sistema para ir al doctor y el proceso que tienes es 
el de toma de los síntomas del paciente, el diagrama BPMN empezaría con paciente que llega al consultorio, después él habla con el médico, el médico lo registra, etc etc, pero todo esto lo vas haciendo con un modelado especial y específico para los BPMN.
mira, te dejo un ss de uno que hice para la práctica
https://user-images.githubusercontent.com/99062142/155254352-4c5d5b4e-0611-4931-a072-e557214026d3.png
Verás que tiene simbología especial, tiene un inicio y un fin y describe los pasos de ese proceso de negocio, pero lo hace empezando con un verbo ya sea en infinitivo o presente y así se va.
estos diagramas son buenos para entender cómo es que se debe comportar el sistema, los pasos que debe de tener, lo que hay que validar, etc. Ligeramente parecido al diagrama de actividades.

**Reply by GloriaCoralCerecedo** [reply]
Oh ya si hay  bastantes, es por eso que me confunde porque son varios en el diseño de la arquitectura de solución, entonce entendí que es el proceso  para realizar demandas. ¿Que software manejas?

**Reply by GabrielaEsquivel** [reply]
Esta vez los hice con LuccidChart, hay otro que me gusta, pero no recuerdo cómo se llama... está padre también y es específicamente para BPMN

**Reply by GloriaCoralCerecedo** [reply]
Oh ya, también está en miro creo

**Comment by ChanLuis** [comment]
Gracias, yo tenía muchas dudas.
